src/main/java/se/sundsvall/byggrarchiver/integration/messaging/MessagingIntegration.java

```diff
@@ -0,0 +1,112 @@
+package se.sundsvall.byggrarchiver.integration.messaging;
+
+import static generated.se.sundsvall.messaging.MessageStatus.SENT;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Optional.ofNullable;
+import static se.sundsvall.dept44.util.ResourceUtils.asString;
+
+import java.util.Base64;
+import java.util.Map;
+
+import org.apache.commons.text.StringSubstitutor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Component;
+
+import se.sundsvall.byggrarchiver.configuration.EmailProperties;
+import se.sundsvall.byggrarchiver.integration.db.model.ArchiveHistory;
+
+import generated.se.sundsvall.messaging.EmailRequest;
+import generated.se.sundsvall.messaging.EmailSender;
+
+@Component
+@EnableConfigurationProperties(MessagingIntegrationProperties.class)
+public class MessagingIntegration {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MessagingIntegration.class);
+
+    static final String INTEGRATION_NAME = "messaging";
+
+    private final Resource geoTekniskHandlingHtmlTemplate = new ClassPathResource("html-templates/geoteknisk_handling_template.html");
+    private final Resource missingExtensionHtmlTemplate = new ClassPathResource("html-templates/missing_extension_template.html");
+
+    private final MessagingClient client;
+    private final EmailProperties emailProperties;
+
+    public MessagingIntegration(final MessagingClient client, final EmailProperties emailProperties) {
+        this.client = client;
+        this.emailProperties = emailProperties;
+    }
+
+    public void sendExtensionErrorEmail(final ArchiveHistory archiveHistory) {
+        var sender = new EmailSender();
+        sender.setName("ByggrArchiver");
+        sender.setAddress(emailProperties.extensionError().sender());
+
+        var values = Map.of(
+            "byggrCaseId", ofNullable(archiveHistory.getCaseId()).orElse(""),
+            "documentName", ofNullable(archiveHistory.getDocumentName()).orElse(""),
+            "documentType", ofNullable(archiveHistory.getDocumentType()).orElse("")
+        );
+        var htmlMessage = toBase64(replace(asString(missingExtensionHtmlTemplate), values));
+
+        var emailRequest = new EmailRequest();
+        emailRequest.setSender(sender);
```

Comment:
Builderpattern?

Reply:
Genererad klientkod, så nja

Reply:
Is something blocking this PR from merge?